File: src/main/java/org/alham/alhamfirst/service/orchestrator/stat/UserStatService.java
package org.alham.alhamfirst.service.orchestrator.stat;

import org.alham.alhamfirst.document.stat.UserStatDocument;
import org.alham.alhamfirst.dto.stat.UserStatDTO;

import java.util.Map;

public interface UserStatService {

    public UserStatDocument saveStat(long userId,  Map<String, Double> statData);
    public UserStatDTO findByUserId(long userId);

    public UserStatDTO findByEncryptedId(String encryptedId);

    public UserStatDTO updateUserStat(long userId, Map<String, Double> statData);


}

File: src/main/java/org/alham/alhamfirst/service/orchestrator/stat/UserStatService.kt
package org.alham.alhamfirst.service.orchestrator.stat;

import org.alham.alhamfirst.document.stat.UserStatDocument;
import org.alham.alhamfirst.dto.stat.UserStatDTO;


interface UserStatService {
    fun createUserStatDocument(userId: Long): UserStatDocument
    fun saveStat(userId: Long, statData: Map<String,Double>): UserStatDocument
    fun findByUserId(userId: Long): UserStatDTO
    fun findByEncryptedId(encryptedId: String): UserStatDTO
    fun updateUserStat(userId: Long, statData: Map<String,Double>, completed: Boolean): UserStatDTO

}
